### PR TITLE
Queues: use case statements in FIFO control

### PR DESCRIPTION
**Pull request by anshumanmohan**
This PR attempts to reap the rewards of #2168 by using case statements in the FIFO. Sadly something seems to be wrong... I am getting an empty `par` in my control instead of the equality checks I would expect. Paging @nathanielnrn to take a look. And of course please let me know if I've made a silly mistake.

In the first commit I massaged my logic to strictly check a cell, `cmd`, for four cases of equality. This passed all queue tests.

Then I tried to move to the `case` idiom in the most obvious way. Sadly I got errors about unused groups. Upon inspection of the `futil` code, which I am also checking in temporarily, it seems the errors are warranted: control is just an empty `par`!
https://github.com/calyxir/calyx/blob/a5a8616d56bb417abb88cef636cf2dca7cbadd63/calyx-py/test/correctness/queues/fifo.futil#L100-L104

### CODE REVIEW

**Comment by anshumanmohan** [comment]
The fix in [94e19ec](https://github.com/calyxir/calyx/commit/94e19ece12c6bd52aeb79ed42a5ebad1a7d12f87) saved the day! Thanks @nathanielnrn!

File: calyx-py/test/correctness/queues/fifo.py [hunk 1]
```diff
@@ -69,14 +82,17 @@ def insert_fifo(prog, name, queue_len_factor=QUEUE_LEN_FACTOR):
...
-        # The logic is shared for pops and peeks, with just a few differences.
-        # Did the user call pop/peek?
-        cb.if_with(fifo.lt_use(cmd, 2), pop_peek_logic),
-        # Did the user call push?
-        cb.if_with(fifo.eq_use(cmd, 2), push_logic),
-        # Did the user call an invalid command?
-        cb.if_with(fifo.eq_use(cmd, 3), raise_err),
+        # Was it a pop, peek, push, or an invalid command?
+        # We can do those four cases in parallel.
+        fifo.case(
```
Comment (polybeandip) [inline calyx-py/test/correctness/queues/fifo.py:87]: [🤯](https://emojipedia.org/exploding-head)

**Review by polybeandip — APPROVED** [review]
Nathaniel's new case statement in the eDSL looks great! These changes to the FIFO really make it look much more crisp.

Minor nitpick: maybe we shouldn't merge `fifo.futil` into main? I understand why you've included it in the PR (so you can show the empty `par` block), but it's not being used to test the FIFO, correct?

**Comment by anshumanmohan** [comment]
Ah yes thanks, I checked it in to show Nathaniel the issue but then forgot to get rid of it. I'll do that and then enable auto-merge

**Pull request merged**